### PR TITLE
SNS - GPIO 

### PR DESCRIPTION
**Pull request by Knhwong**

### CODE REVIEW

**Review by kshxtij — COMMENTED** [review]
Why were the files within scripts/hooks removed?

File: lib/io/gpio.cpp [hunk 1]
```diff
@@ -1,22 +1,289 @@
...
+
+  char buf[10];
+  for (uint32_t pin : exported_pins) {
+    snprintf(buf, sizeof(buf), "%d", pin);
+    write(fd, buf, strlen(buf) + 1);
+  }
+  close(fd);
 }
+//Pins must be set before we export and attach
+void Gpio::exportGPIO()
```
Comment (mifrandir) [inline lib/io/gpio.cpp:68]: exportGpio

File: lib/io/gpio.cpp [hunk 2]
```diff
@@ -1,22 +1,289 @@
...
+    if (fdset.revents & POLLERR) { log_.error("an error on wait of gpio %d: %d", pin_, errno); }
+  }
+
+  log_.error("some error on wait of gpio %d", pin_);
+  return -1;
+}
+
+
+
+uint8_t Gpio::Write(const uint8_t pin, const bool SetOrClear) {
```
Comment (mifrandir) [inline lib/io/gpio.cpp:191]: Gpio::write

**Review by mifrandir — CHANGES_REQUESTED** [review]
Id get rid of most of this old interface. What was so bad about the dummy that was there before?

**Comment by Knhwong** [comment]
#34

**Pull request closed without merging**